daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/ElementCombinator1.scala

```diff
@@ -199,7 +199,8 @@ abstract class ElementParserBase(
       if (pstate.processorStatus ne Success) return
 
       val shouldValidate = erd.isSimpleType &&
-        pstate.dataProc.isDefined && pstate.dataProc.value.validationMode != ValidationMode.Off
+        pstate.dataProc.isDefined && (pstate.dataProc.value.validationMode != ValidationMode.Off
+          && pstate.dataProc.value.validationMode != ValidationMode.Xerces)
```

Comment:
Is this correct here as it seems that shouldValidate will always be false if ValidationMode.Xerces.  Is there a reason we need it to be false when validating with Xerces here?

Also this might be another case where a match statement makes things more clear.

Reply:
Yes, this shouldValidate is what controls whether we do limited validation, so this condition is saying do limited validation when validation mode is limited and full!